rbc/tests/test_omnisci_array_methods.py

```diff
@@ -0,0 +1,103 @@
+import pytest
+import numpy as np
+
+
+rbc_omnisci = pytest.importorskip('rbc.omniscidb')
+available_version, reason = rbc_omnisci.is_available()
+pytestmark = pytest.mark.skipif(not available_version, reason=reason)
+
+
+@pytest.fixture(scope='module')
+def omnisci():
+    config = rbc_omnisci.get_client_config(debug=not True)
+    m = rbc_omnisci.RemoteOmnisci(**config)
+    yield m
+
+
+def test_ndarray_methods(omnisci):
+    omnisci.reset()
+    from rbc.omnisci_array import Array
+
+    @omnisci('double(int64, double)')
+    def ndarray_fill(size, v):
+        a = Array(size, 'double')
+        a.fill(v)
+        # XXX currently it's not possible to return the array
```

Comment:
This is not true any more and one can return `a`. Update the test function accordingly.

Reply:
Update the test and it seems there is a bug in omnisci server. See:

```
args_notnull:                                     ; preds = %groupby_nullcheck_true
  %11 = sext i32 %arg_literal_0 to i64
  %12 = call double @cast_int64_t_to_double_nullable(i64 %arg_literal_8, i64 -9223372036854775808, double 0x10000000000000)
  %13 = fdiv double %12, 1.000000e+01
  %14 = alloca { double*, i64, i8 }
  call void @ndarray_fill__0({ double*, i64, i8 }* %14, i64 %11, double %13)
  br label %args_null

args_null:                                        ; preds = %args_notnull, %groupby_nullcheck_true
  %15 = phi { double*, i64, i8 }* [ %14, %args_notnull ], [ %5, %groupby_nullcheck_true ]
  %16 = getelementptr inbounds { double*, i64, i8 }, { double*, i64, i8 }* %5, i32 0, i32 2
 >> store i1 true, i8* %16  // <------ Here! It's trying to store an int1 into a int8 pointer
  %17 = getelementptr inbounds { double*, i64, i8 }, { double*, i64, i8 }* %5, i32 0, i32 1
  store i64 0, i64* %17
  %18 = load { double*, i64, i8 }, { double*, i64, i8 }* %15
  %19 = extractvalue { double*, i64, i8 } %18, 0
  %20 = bitcast double* %19 to i8*
  %21 = extractvalue { double*, i64, i8 } %18, 1
  %22 = extractvalue { double*, i64, i8 } %18, 2
  %23 = icmp eq i8 %22, 1
  br i1 %23, label %arr_nullcheck_fail_bb, label %arr_nullcheck_ok_bb
```

This error happens if one calls the test with `(5, 4)` as inputs. The second input is an integer but the test was declared expecting it to be a `double`. Anything we can do here?

Reply:
I cannot reproduce. The server generated LLVM IR is here:
```
...
groupby_nullcheck_true:                           ; preds = %filter_true
  %5 = sext i32 %arg_literal_0 to i64
  %6 = sitofp i32 %arg_literal_4 to double
  %7 = alloca { double*, i64, i8 }
  call void @ndarray_fill__0({ double*, i64, i8 }* %7, i64 %5, double %6)
  %8 = load { double*, i64, i8 }, { double*, i64, i8 }* %7
  %9 = extractvalue { double*, i64, i8 } %8, 0
  %10 = bitcast double* %9 to i8*
  %11 = extractvalue { double*, i64, i8 } %8, 1
  %12 = extractvalue { double*, i64, i8 } %8, 2
  %13 = icmp eq i8 %12, 1
  br i1 %13, label %arr_nullcheck_fail_bb, label %arr_nullcheck_ok_bb

groupby_nullcheck_false:                          ; preds = %filter_true
  %14 = trunc i64 %pos to i32
  %15 = sub i32 0, %14
  ret i32 %15

arr_nullcheck_ok_bb:                              ; preds = %groupby_nullcheck_true
  call void @register_buffer_with_executor_rsm(i64 139901849394096, i8* %10)
  br label %arr_return

arr_nullcheck_fail_bb:                            ; preds = %groupby_nullcheck_true
  br label %arr_return

arr_return:                                       ; preds = %arr_nullcheck_fail_bb, %arr_nullcheck_ok_bb
  %array_ptr_return = phi i8* [ %10, %arr_nullcheck_ok_bb ], [ null, %arr_nullcheck_fail_bb ]
  %agg_col_ptr = getelementptr i64, i64* %3, i32 0
  %16 = ptrtoint i8* %array_ptr_return to i64
  call void @agg_id(i64* %agg_col_ptr, i64 %16)
  %agg_col_ptr1 = getelementptr i64, i64* %3, i32 1
  call void @agg_id(i64* %agg_col_ptr1, i64 %11)
  br label %filter_false
}
```

This looks different from your report. What omniscidb version are you using?

Reply:
> What omniscidb version are you using?
Master. I just compiled a recent version, and the same error happened.

Apply this diff and see if you have the same error
```
diff --git a/rbc/tests/test_omnisci_array_methods.py b/rbc/tests/test_omnisci_array_methods.py
index c2532c5..2d947ec 100644
--- a/rbc/tests/test_omnisci_array_methods.py
+++ b/rbc/tests/test_omnisci_array_methods.py
@@ -99,7 +99,7 @@ def test_ndarray_methods(omnisci):
         return a.prod(initial=initial)

     ndarray_methods = [
-        ('fill', (5, 4), [4.0, 4.0, 4.0, 4.0, 4.0]),
+        ('fill', (5, 4.0), [4.0, 4.0, 4.0, 4.0, 4.0]),
         ('max', (5, 4.0), 4.0),
         ('max_empty', (0, ), -128),
         ('max_initial', (5, 4.0, 30.0), 30.0),
```

Reply:
Opened https://github.com/xnd-project/rbc/issues/81